saleor/product/views.py

```diff
@@ -112,8 +114,17 @@ def product_add_to_cart(request, slug, product_id):
 def category_index(request, path, category_id):
     category = get_object_or_404(Category, id=category_id)
     actual_path = category.get_full_path()
+    products = products_with_details(
+        user=request.user).filter(categories__name=category)
+    product_filter = ProductFilter(request.GET, queryset=products)
+    products = [
+        (product, get_availability(product, request.discounts, request.currency))
+        for product in product_filter.qs]
     if actual_path != path:
         return redirect('product:category', permanent=True, path=actual_path,
                         category_id=category_id)
-    return TemplateResponse(request, 'category/index.html',
-                            {'category': category})
+    products = get_paginator_items(
```

Comment:
Reusing same symbol in local scope for objects of different types is unclear and a bit confusing. I would use separate names for subsequent products, even extract some of them since this function is clearly too long and does more than one thing.

Reply:
Done but if i am correct we have this style in every view with paginated items.